Schema guide descriptions for root-level keys abstract through message (#242)
citation-file-format/citation-file-format: schema-guide.md

### `contact`

- type: Array of [`definitions.person`](#definitionsperson) and/or [`definitions.entity`](#definitionsentity) objects.
- required: `false`

The contact person, group, company, etc. for the software or dataset.

Usage examples:

```yaml
contact:
  - affiliation: "Humboldt-Universität zu Berlin"
    email: "mail@sdruskat.net"
    family-names: Druskat
    given-names: Stephan
```
```yaml
contact:
  - email: "mail@research-project.org"
    name: "The Research Software project"
```
```yaml
contact:
  - email: "mail@sdruskat.net"
    given-names: Stephan
    family-names: Druskat
  - email: "mail@research-project.org"
    name: "The Research Software project"
```

### `date-released`

- type: [`definitions.date`](#definitionsdate)
- required: `false`

The date the software or data set has been released. Format is 4-digit year, 2-digit month, 2-digit day of month, separated by dashes.

Usage example:

```yaml
date-released: 2020-01-31
```

### `doi`

- type: [`definitions.doi`](#definitionsdoi)
- required: `false`

> The DOI of the software or data set. This notation is most useful when there is just one DOI you want to include. In

Reviewer: ```suggestion
The DOI of the software or dataset. This notation is most useful when there is just one DOI you want to include. In
```
See https://github.com/citation-file-format/citation-file-format/pull/242/files#r676562871

that case, `doi` can be used as shorthand for something like:

```yaml
identifiers:
  - type: doi
    value: 10.5281/zenodo.1003149
    description: The concept DOI of the work.
```
or

```yaml
identifiers:
  - type: doi
    value: 10.5281/zenodo.4813122
    description: The versioned DOI of the work.
```

Usage examples:

```yaml
doi: 10.5281/zenodo.1003149
```
```yaml
doi: 10.5281/zenodo.4813122
```

### `identifiers`

- type: array of [`definitions.identifier`](#definitionsidentifier) objects.
- required: `false`

The identifiers of the software or dataset.

Usage example: see [`definitions.identifier`](#definitionsidentifier).

### `keywords`

- type: array of `string`
- required: `false`

Keywords that describe the work.

Usage example:

```yaml
keywords:
 - thefirstkeyword
 - thesecondkeyword

>  - and a third

Reviewer: ```suggestion
 - a third keyword
```

Just to make this complete and in line with the two examples above, taken this is meant to show that you can use a string with spaces here.

- required: `false`

The SPDX license identifier(s) for the license(s) under which the work is made available. When there are multiple
licenses, it is assumed their relationship is OR, not AND.

Usage examples:

```yaml
license: Apache-2.0
```
```yaml
license:
 - Apache-2.0
 - MIT
```
```yaml
license:
 - GPL-3.0
 - GPL-3.0-or-later
```

### `license-url`

- type: [`definitions.url`](#definitionsurl)
- required: `false`

The URL of the license text under which the software or dataset is licensed (only for non-standard licenses not included in the SPDX License List).

Usage example:

```yaml
license-url: "https://obscure-licenses.com?id=1234"
```

### `message`

- type: `string`
- required: `true`
- default: `If you use this software, please cite it using the metadata from this file.`

A message to the human reader of the CITATION.cff file to let them know what to do with the citation metadata.

Usage examples:

```yaml
message: If you use this software, please cite it using the metadata from this file.
```
```yaml
message: Please cite this software using these metadata.
```
```yaml
message: Please cite this software using the metadata from 'preferred-citation'.
```
```yaml

> message: If you use this data set, please cite it using the metadata from this file.

Reviewer: ```suggestion
message: If you use this dataset, please cite it using the metadata from this file.
```

[CED](https://dictionary.cambridge.org/dictionary/english/dataset) says one word, although [APA Style Blog](https://blog.apastyle.org/apastyle/2012/07/data-is-or-data-are.html) says differently (but who are they in the world of lexicographers!), so I asked the world corpus Google Search which gives 64M hits for `"data set"`and 125M for `"dataset"`. Clear result if you ask me.

Author: good enough for me :)

```
```yaml

> message: Please cite this data set using these metadata.

Reviewer: ```suggestion
message: Please cite this dataset using these metadata.
```

See https://github.com/citation-file-format/citation-file-format/pull/242/files#r676562871

```
```yaml

> message: Please cite this data set using the metadata from 'preferred-citation'.

Reviewer: ```suggestion
message: Please cite this dataset using the metadata from 'preferred-citation'.
```

See https://github.com/citation-file-format/citation-file-format/pull/242/files#r676562871

The entity's fax number.

Usage example:

```yaml
authors:
  - name: The Research Software Project

>     fax: +012-3456-7890

Reviewer: ```suggestion
    fax: +12-3456-7890
```

Just an OCD alert, the `+` replaces the intl `00` "Verkehrsausscheidungsziffer" (there isn't even an English Wikipedia page for telephone country codes 😱, the related Dutch one calls this simply "[Landnummer](https://nl.wikipedia.org/wiki/Landnummer)" I think), and there are countries without the `0` prefix, so <wince> can we not use this here?

The entity's telephone number.

Usage example:

```yaml
authors:
  - name: The Research Software Project

>     tel: +012-345-6789098

Reviewer: ```suggestion
    tel: +12-345-6789098
```

See https://github.com/citation-file-format/citation-file-format/pull/242/files#r676568214

The entity's website.

Usage example:

```yaml
authors:
  - name: The Research Software Project
    website: https://research-software-project.org
```

### `definitions.fax`

`definitions` and its subkeys should not be used directly in `CITATION.cff` files.

- type: `string`
- required: `false`

A fax number.

Usage example:

```yaml
authors:
  - name: The Research Software Project

>     fax: +012-3456-7890

Reviewer: ```suggestion
    fax: +12-3456-7890
```

See https://github.com/citation-file-format/citation-file-format/pull/242/files#r676568214

```
```yaml
authors:
  - family-names: McClane

>     fax: +012-3456-7890

Reviewer: ```suggestion
    fax: +12-3456-7890
```

https://github.com/citation-file-format/citation-file-format/pull/242/files#r676568214

    given-names: John
```

### `definitions.identifier`

`definitions` and its subkeys should not be used directly in `CITATION.cff` files.

- type: `...`
- required: `false`

...

Usage examples:

```yaml
identifiers:
  - type: doi
    value: 10.5281/zenodo.1003149
    description: The concept DOI of the work.
```
```yaml
identifiers:
  - type: doi
    value: 10.5281/zenodo.4813122
    description: The versioned DOI for version 1.1.0 of the work.
```
```yaml
identifiers:
  - type: doi
    value: 10.5281/zenodo.1003149
    description: The concept DOI of the work.
  - type: doi
    value: 10.5281/zenodo.4813122
    description: The versioned DOI for version 1.1.0 of the work.
```
```yaml
identifiers:
  - type: doi
    value: 10.5281/zenodo.1003149
    description: The concept DOI of the work.
  - type: doi
    value: 10.5281/zenodo.4813122
    description: The versioned DOI for version 1.1.0 of the work.
  - type: swh
    value: swh:1:dir:bc286860f423ea7ced246ba7458eef4b4541cf2d
    description: The Software Heritage identifier for version 1.1.0 of the work.
  - type: url
    value: https://github.com/citation-file-format/citation-file-format/releases/tag/1.1.0

>     description: The GitHub release URL to tag 1.1.0.

Reviewer: ```suggestion
    description: The GitHub release URL of tag 1.1.0.
```

IMHO it should be "locator of" ...

  - type: url
    value: https://github.com/citation-file-format/citation-file-format/tree/16192bf05e99bcb35d5c3e085047807b5720fafc

>     description: The GitHub release URL to the commit of tag 1.1.0.

Reviewer: ```suggestion
    description: The GitHub release URL to the commit of tag 1.1.0.
```

See https://github.com/citation-file-format/citation-file-format/pull/242/files#r676571683

`definitions` and its subkeys should not be used directly in `CITATION.cff` files.

- type: (array of) [`definitions.license-enum`](#definitions.license-enum) objects.
- required: `false`

License identifier(s) under which the work is made available. When there are multiple licenses, it is assumed their relationship is OR, not AND.

Usage examples:

```yaml
license: Apache-2.0
```
```yaml
license:
  - Apache-2.0
  - MIT
```

### `definitions.license-enum`

`definitions` and its subkeys should not be used directly in `CITATION.cff` files.

- type: `enum` with values:

>     - `0BSD`

Reviewer: No way around this big list I guess. We could contract into a comma-separated list to save screen space, but this would probably make it less usable, what do you think?

Author: yeah there isnt really a good way of dealing with this. Maybe a table could work, but I think I'm right in saying you cant have a table nested in a list. Alternatively, we could do a fixed-width font multicolumn print, like what you would see in a terminal, to help people not injure themselves scrolling through the license list.

`definitions` and its subkeys should not be used directly in `CITATION.cff` files.

> - type: - type: [`definitions.alias`](#definitionsalias)

Reviewer: ```suggestion
- type: [`definitions.alias`](#definitionsalias)
```

Remove duplicate `- type: ` ...

`definitions` and its subkeys should not be used directly in `CITATION.cff` files.

- type: [`definitions.fax`](#definitionsfax)
- required: `false`

The person's fax number.

Usage example:

```yaml
authors:
  - family-names: McClane

>     fax: +012-3456-7890

Reviewer: ```suggestion
    fax: +12-3456-7890
```

See https://github.com/citation-file-format/citation-file-format/pull/242/files#r676568214

`definitions` and its subkeys should not be used directly in `CITATION.cff` files.

- type: `string`
- required: `false`

> The person's name particle. The person's name particle, e.g., a nobiliary particle or a preposition meaning 'of' or 'from' (for example 'von' in 'Alexander von Humboldt').

Author: ```suggestion
The person's name particle, e.g., a nobiliary particle or a preposition meaning 'of' or 'from' (for example 'von' in 'Alexander von Humboldt').
```

`definitions` and its subkeys should not be used directly in `CITATION.cff` files.

- type: `string`
- required: `false`

> The person's name suffix. The person's name-suffix, e.g. 'Jr.' for Sammy Davis Jr. or 'III' for Frank Edwin Wright III.

Author: ```suggestion
The person's name-suffix, e.g. 'Jr.' for Sammy Davis Jr. or 'III' for Frank Edwin Wright III.
```

`definitions` and its subkeys should not be used directly in `CITATION.cff` files.

- type: [`definitions.tel`](#definitionstel)
- required: `false`

The person's telephone number.

Usage example:

```yaml
authors:
  - family-names: McClane
    given-names: John

>     tel: +012-345-6789098

Reviewer: ```suggestion
    tel: +12-345-6789098
```

See https://github.com/citation-file-format/citation-file-format/pull/242/files#r676568214

`definitions` and its subkeys should not be used directly in `CITATION.cff` files.

- type: `string`
- required: `false`

A telephone number.

Usage examples:

```yaml
authors:
  - family-names: McClane
    given-names: John

>     tel: +012-345-6789098

Reviewer: ```suggestion
    tel: +12-345-6789098
```

See https://github.com/citation-file-format/citation-file-format/pull/242/files#r676568214

```
```yaml
authors:
  - name: The Research Software Project

>     tel: +012-345-6789098

Reviewer: ```suggestion
    tel: +12-345-6789098
```

See https://github.com/citation-file-format/citation-file-format/pull/242/files#r676568214

```

### `definitions.url`

`definitions` and its subkeys should not be used directly in `CITATION.cff` files.

- type: `string`
- required: `false`

The URL of a landing page/website for the software or dataset. Supported URLs start with one of:

- `https://`
- `http://`
- `ftp://`
- `sftp://`

Usage examples:

```yaml
url: "https://citation-file-format.github.io/"
```
```yaml
authors:
  - name: The Research Software Project
    url: "https://research-software-project.org"
```
```yaml
references:

>   -
>     name: The Research Software Project

Reviewer: ```suggestion
  - name: The Research Software Project
```
While this would've been valid YAML AFAIK, put on one line to save space (and let people figure out their YAML by reading *their* specs).
